test/integration/scheduler/framework_test.go

```diff
@@ -84,12 +92,37 @@ const (
 )
 
 var _ = framework.PrefilterPlugin(&PrefilterPlugin{})
+var _ = framework.FilterPlugin(&FilterPlugin{})
 var _ = framework.ReservePlugin(&ReservePlugin{})
 var _ = framework.PrebindPlugin(&PrebindPlugin{})
 var _ = framework.PostbindPlugin(&PostbindPlugin{})
 var _ = framework.UnreservePlugin(&UnreservePlugin{})
 var _ = framework.PermitPlugin(&PermitPlugin{})
 
+// Name returns name of the plugin.
+func (rp *FilterPlugin) Name() string {
+	return filterPluginName
+}
+
+var filterPlugin = &FilterPlugin{}
+
+// Filter is a test function that returns an error or nil, depending on the
+// value of "failFilter".
+func (rp *FilterPlugin) Filter(pc *framework.PluginContext, pod *v1.Pod, nodeInfo *schedulernodeinfo.NodeInfo) *framework.Status {
+	rp.numFilterCalled++
+
+	if rp.failFilter {
+		return framework.NewStatus(framework.Error, fmt.Sprintf("injecting failure for pod %v", pod.Name))
+	}
+
+	return framework.NewStatus(framework.Success, "")
+}
+
```

Comment:
Please add a reset method for FilterPlugin and call it after each test. You could follow the pattern in the current master branch.